src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectSnapshotManagerBase.cs

```diff
@@ -18,31 +18,31 @@ internal abstract class ProjectSnapshotManagerBase : ProjectSnapshotManager
 
     internal abstract IReadOnlyCollection<string> OpenDocuments { get; }
 
-    internal abstract void DocumentAdded(HostProject hostProject, HostDocument hostDocument, TextLoader textLoader);
+    internal abstract void DocumentAdded(ProjectKey projectKey, HostDocument hostDocument, TextLoader textLoader);
 
-    internal abstract void DocumentOpened(string projectFilePath, string documentFilePath, SourceText sourceText);
+    internal abstract void DocumentOpened(ProjectKey projectKey, string documentFilePath, SourceText sourceText);
 
-    internal abstract void DocumentClosed(string projectFilePath, string documentFilePath, TextLoader textLoader);
+    internal abstract void DocumentClosed(ProjectKey projectKey, string documentFilePath, TextLoader textLoader);
 
-    internal abstract void DocumentChanged(string projectFilePath, string documentFilePath, TextLoader textLoader);
+    internal abstract void DocumentChanged(ProjectKey projectKey, string documentFilePath, TextLoader textLoader);
 
-    internal abstract void DocumentChanged(string projectFilePath, string documentFilePath, SourceText sourceText);
+    internal abstract void DocumentChanged(ProjectKey projectKey, string documentFilePath, SourceText sourceText);
 
-    internal abstract void DocumentRemoved(HostProject hostProject, HostDocument hostDocument);
+    internal abstract void DocumentRemoved(ProjectKey projectKey, HostDocument hostDocument);
 
     internal abstract void ProjectAdded(HostProject hostProject);
 
     internal abstract void ProjectConfigurationChanged(HostProject hostProject);
```

Comment:
I've always wondered why `ProjectConfigurationChanged` didn't use the same key as other methods

Reply:
Yeah, I don't know. I think it could just take a key, a configuration, and a namespace.